### ISSUE TITLE
Improve Initial Checkpoint Load onto CPU/GPU

### ISSUE THREAD

**Issue by ryan13mt** [painebenjamin/app.enfugue.ai#27]
Hey @painebenjamin it's me again sorry. I installed the latest update and got the following logs, maybe they can help you think of possible issues that might be causing this. 

This is how i setup the model:
![image](https://github.com/painebenjamin/app.enfugue.ai/assets/6053618/aa844f11-47ca-433b-84f7-096aaab666c1)

And the log files with only the failing call:
[enfugue.log](https://github.com/painebenjamin/app.enfugue.ai/files/11913925/enfugue.log)
[enfugue-engine.log](https://github.com/painebenjamin/app.enfugue.ai/files/11913926/enfugue-engine.log)

**Comment by painebenjamin** [comment]
Don't be sorry at all, you've saved me hours and hours of chasing down bugs already. So thank you!

After investigating how A1111 handles this, it looks like it loads the initial checkpoint on the CPU first. I'm loading it directly onto the GPU straight away - which is probably a little faster, but for a gigantic checkpoint like SD 1.5 base, it overloads your VRAM.

Theirs:
![image](https://github.com/painebenjamin/app.enfugue.ai/assets/57536852/8dc97ab7-58b7-425f-9f85-d8f6c4718000)

Mine:
![image](https://github.com/painebenjamin/app.enfugue.ai/assets/57536852/8abc2b4c-fdb8-460b-b3f4-1ef4a441280a)

I'm going to investigate setting this to always load to CPU first and see how it affects performance. If it's not too bad, I'll leave it as always CPU - if it dips performance too much, I'll add some logic around comparing the size of the checkpoint and the size of your available VRAM before loading it.

If you're willing to try something while I'm working - have you tried using a more fine-tuned checkpoint off of CivitAI or elsewhere? Most of those clock in at around 2GB, which is way smaller than the base models' ~7GB. If my guess is correct, those should be able to work for you.

**Comment by ryan13mt** [comment]
I tried it and i'm happy to share my first generation using Enfugue. Thanks a lot for your work, this is miles easier to use compared to Auto1111. Dont get me wrong, it has the most functionality but it takes time to setup.

I used the pruned Dreamshaper model for this.
Would be nice to have the functionality to use the full ones later on.

![image](https://github.com/painebenjamin/app.enfugue.ai/assets/6053618/ed9424e1-de19-4204-9641-4bc984a87a39)
